src/olympia/blocklist/admin.py

```diff
@@ -696,7 +697,17 @@ def formfield_for_dbfield(self, db_field, request, **kwargs):
     def changeform_view(self, request, obj_id=None, form_url='',
                         extra_context=None):
         extra_context = extra_context or {}
-        obj = self.get_object(request, obj_id) if obj_id else None
+        if obj_id:
+            obj = (
+                self.get_object(request, obj_id) or
+                # if we can't find the obj_id maybe it's a guid instead
+                self.get_object(request, obj_id, 'guid'))
+            if obj and str(obj_id) != str(obj.id):
+                # we found it from the guid if the obj_id != obj.id so redirect
+                url = request.path.replace(obj_id, str(obj.id), 1)
+                return http.HttpResponsePermanentRedirect(url)
```

Comment:
Could we avoid that redirect if we just set

```
obj_id = obj.id
```

here?

Reply:
we could, but we want the redirect so there are no weird edge cases.